presto-main/src/main/java/com/facebook/presto/spiller/GenericPartitioningSpiller.java

```diff
@@ -46,10 +47,15 @@
         implements PartitioningSpiller
 {
     private final List<Type> types;
-    private final PageBuilder[] pageBuilders;
     private final PartitionFunction partitionFunction;
-    private final SingleStreamSpiller[] spillers;
     private final Closer closer = Closer.create();
+    private final SingleStreamSpillerFactory spillerFactory;
```

Comment:
The randomization works great for large numbers, but e.g. in the case of  a 2 disk setup with task concurrency 4 i have 62.5% chance for a skew in disk usage. I decided not to add randomization in `FSSSFactory`, but i can still be convinced otherwise.